src/custom_nodes/object_detection/object_detection.cpp

```diff
@@ -0,0 +1,340 @@
+    uint64_t maxOutputBatch = get_int_parameter("max_output_batch", params, paramsCount, 100);
+    NODE_ASSERT(maxOutputBatch > 0, "max output batch must be larger than 0");
+    int filterLabelId = get_int_parameter("filter_label_id", params, paramsCount, -1);
+    bool debugMode = get_string_parameter("debug", params, paramsCount) == "true";
+
+    const CustomNodeTensor* imageTensor = nullptr;
+    const CustomNodeTensor* detectionTensor = nullptr;
+
+    for (int i = 0; i < inputsCount; i++) {
+        if (std::strcmp(inputs[i].name, INPUT_IMAGE_TENSOR_NAME) == 0) {
+            imageTensor = &(inputs[i]);
+        } else if (std::strcmp(inputs[i].name, INPUT_DETECTION_TENSOR_NAME) == 0) {
+            detectionTensor = &(inputs[i]);
+        } else {
+            std::cout << "Unrecognized input: " << inputs[i].name << std::endl;
+            return 1;
+        }
+    }
+
+    NODE_ASSERT(imageTensor != nullptr, "Missing input image");
+    NODE_ASSERT(detectionTensor != nullptr, "Missing input scores");
+    NODE_ASSERT(imageTensor->precision == FP32, "image input is not FP32");
+    NODE_ASSERT(detectionTensor->precision == FP32, "image input is not FP32");
+
+    NODE_ASSERT(imageTensor->dimsCount == 4, "input image shape must have 4 dimensions");
+    NODE_ASSERT(imageTensor->dims[0] == 1, "input image batch must be 1");
+    NODE_ASSERT(imageTensor->dims[1] == 3, "input image needs to have 3 color channels");
+
+    NODE_ASSERT(detectionTensor->dimsCount == 4, "input detection shape must have 4 dimensions");
+    NODE_ASSERT(detectionTensor->dims[0] == 1, "input detection dim[0] must be 1");
+    NODE_ASSERT(detectionTensor->dims[1] == 1, "input detection dim[1] must be 1");
+    NODE_ASSERT(detectionTensor->dims[2] == 200, "input detection dim[2] must be 200");
+    NODE_ASSERT(detectionTensor->dims[3] == 7, "input detection dim[3] must be 7");
+
+    uint64_t _imageHeight = imageTensor->dims[2];
+    uint64_t _imageWidth = imageTensor->dims[3];
+    NODE_ASSERT(_imageHeight <= std::numeric_limits<int>::max(), "image height is too large");
+    NODE_ASSERT(_imageWidth <= std::numeric_limits<int>::max(), "image width is too large");
+    int imageHeight = static_cast<int>(_imageHeight);
+    int imageWidth = static_cast<int>(_imageWidth);
+
+    if (debugMode) {
+        std::cout << "Processing input tensor image resolution: " << cv::Size(imageHeight, imageWidth) << "; expected resolution: " << cv::Size(originalImageHeight, originalImageWidth) << std::endl;
+    }
+
+    NODE_ASSERT(imageHeight == originalImageHeight, "original image size parameter differs from original image tensor size");
+    NODE_ASSERT(imageWidth == originalImageWidth, "original image size parameter differs from original image tensor size");
+
+    cv::Mat image = nchw_to_mat(imageTensor);
+    // cv::Mat image = nhwc_to_mat(imageTensor);  // here you can support other layouts
```

Comment:
Do we want to leave this comment?

Reply:
Yes, since this is an example, it would be good to give some hints.